dsl_parser/functions.py

```diff
@@ -147,25 +147,83 @@ def evaluate_runtime(self, storage):
 class GetInput(Function):
 
     def __init__(self, args, **kwargs):
-        self.input_name = None
+        self.input_value = None
         super(GetInput, self).__init__(args, **kwargs)
 
     def parse_args(self, args):
-        valid_args_type = isinstance(args, basestring)
-        if not valid_args_type:
+        if not isinstance(args, basestring) \
+                and not (isinstance(args, list) and len(args) >= 1):
             raise ValueError(
-                "get_input function argument should be a string, "
-                "but is '{0}'.".format(args))
-        self.input_name = args
+                "Illegal argument(s) passed to {0} function. Expected either"
+                "a string or a list [input_name [, key1/index1 [,...]]] but "
+                "got {1}".format(self.name, args))
```

Comment:
Maybe ', '.join([str(arg) for arg in args])?

Reply:
Yeah, why not.